### PR TITLE
Mon 12198 improve sql queries by escaping special character

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

**Fixes** MON-12198

Improve sql search performance

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

See how to test section in [jira ticket](https://centreon.atlassian.net/jira/software/c/projects/MON/boards/113?modal=detail&selectedIssue=MON-12198&sprint=654)

### CODE REVIEW

**Review by adr-mo — DISMISSED** [review]

**Review by sc979 — DISMISSED** [review]

File: www/include/monitoring/status/Notifications/notifications.php [hunk 1]
```diff
@@ -60,7 +60,7 @@
 if (!isset($_SESSION['centreon_notification_preferences'])) {
     $userId = $centreon->user->get_id();
     $resPref = $obj->DB->query("SELECT cp_key, cp_value
-         FROM contact_param 
+         FROM contact_param
          WHERE cp_key LIKE 'monitoring%notification%'
          AND cp_contact_id = '" . $obj->DB->escape($userId) . "'");
```
Comment (sc979) [inline www/include/monitoring/status/Notifications/notifications.php:65]: kindly, could you bind $userId and sanitize it ?

Comment (AssanDiomande) [reply www/include/monitoring/status/Notifications/notifications.php]: Done

File: www/include/monitoring/status/Notifications/notifications.php [hunk 2]
```diff
@@ -59,18 +59,20 @@
...
+            FROM contact_param
+            WHERE cp_key LIKE 'monitoring%notification%'
+            AND cp_contact_id = :cp_contact_id");
+        $resPref->bindValue(":cp_contact_id", $obj->DB->escape($userId), PDO::PARAM_INT);
+        $resPref->execute();
+        $notificationPreferences = [];
+        while ($rowPref = $resPref->fetch()) {
+            $notificationPreferences[$rowPref['cp_key']] = $rowPref['cp_value'];
+        }
+        $_SESSION['centreon_notification_preferences'] = $notificationPreferences;
```
Comment (sc979) [inline www/include/monitoring/status/Notifications/notifications.php:74]: $resPref = $obj->DB->prepare("SELECT cp_key, cp_value
        FROM contact_param
        WHERE cp_key LIKE 'monitoring%notification%'
        AND cp_contact_id = :cp_contact_id");
    $resPref->bindValue(":cp_contact_id", (int) $userId, PDO::PARAM_INT);
    $resPref->execute();
    $notificationPreferences = [];
    while ($rowPref = $resPref->fetch()) {
        $notificationPreferences[$rowPref['cp_key']] = $rowPref['cp_value'];
    }
        $_SESSION['centreon_notification_preferences'] = $notificationPreferences;

Comment (AssanDiomande) [reply www/include/monitoring/status/Notifications/notifications.php]: done

**Review by sc979 — DISMISSED** [review]

File: www/include/monitoring/status/Notifications/notifications.php [hunk 3]
```diff
@@ -59,10 +59,12 @@
...
     $userId = $centreon->user->get_id();
-    $resPref = $obj->DB->query("SELECT cp_key, cp_value
-         FROM contact_param 
-         WHERE cp_key LIKE 'monitoring%notification%'
-         AND cp_contact_id = '" . $obj->DB->escape($userId) . "'");
+    $resPref = $obj->DB->prepare("SELECT cp_key, cp_value
+        FROM contact_param
+        WHERE cp_key LIKE 'monitoring%notification%'
+        AND cp_contact_id = :cp_contact_id");
+    $resPref->bindValue(":cp_contact_id", (int)$userId, PDO::PARAM_INT);
```
Comment (kduret) [inline www/include/monitoring/status/Notifications/notifications.php:66]:
```suggestion
    $resPref->bindValue(":cp_contact_id", (int) $userId, \PDO::PARAM_INT);
```

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**